### PR TITLE
new methods to expose plt in rendering functions

### PR DESCRIPTION
**Pull request by ronuchit**
now we have: `render_state_plt`, `render_state`, `render_plt`, and `render`, but most environments will only have to define the first, since `render_state` by default calls `utils.fig2data` on the return value of `render_state_plt`, and `render_plt` is to `render_state_plt` as `render` is to `render_state`

a step toward #730

### CODE REVIEW

File: src/envs/base_env.py [hunk 1]
```diff
@@ -87,22 +90,62 @@ def action_space(self) -> Box:
...
+        """Render a state and action into a Matplotlib figure.
+
+        Like simulate, this function is not meant to be part of the
+        "final system", where the environment is the real world. It is
+        just for convenience, e.g., in test coverage.
+
+        For environments which don't use Matplotlib for rendering, this
+        function should be overriden to simply crash.
+
+        NOTE: Users of this method must remember to call `plt.close()`,
```
Comment (tomsilver) [inline src/envs/base_env.py:108]: if you wanna be fancy, we could try something like https://stackoverflow.com/questions/63508841/using-with-statement-in-python-to-auto-close-a-pyplot-figure

File: tests/test_utils.py [hunk 2]
```diff
@@ -5,10 +5,10 @@
 from typing import Iterator, Tuple
 from typing import Type as TypingType
 
+import matplotlib.pyplot as plt
```
Comment (tomsilver) [inline tests/test_utils.py:8]: y?

**Review by tomsilver — APPROVED** [review]
good stuffs

File: tests/test_utils.py (hunk 2 above)
Comment (ronuchit) [reply tests/test_utils.py:8]: just for consistency, i noticed we were doing it both ways

File: src/envs/base_env.py (hunk 1 above)
Comment (ronuchit) [reply src/envs/base_env.py:108]: ehhh

Comment (tomsilver) [reply src/envs/base_env.py:108]: yeah lol nvm

**Pull request merged**